Add the `enabledFeatures` property to WebXR Sessions.

AdaRoseCannon opened the pull request:
#### 2b89d244a43e806e1d15711b74b51e0cc4452537
<pre>
Add the `enabledFeatures` property to WebXR Sessions.
<a href="https://bugs.webkit.org/show_bug.cgi?id=277155">https://bugs.webkit.org/show_bug.cgi?id=277155</a>

Reviewed by Mike Wyrzykowski.

Add the `enabledFeatures` property to WebXR Sessions.

* Source/WebCore/Modules/webxr/WebXRSession.cpp: Modified to add EnabledFeatures Property
* Source/WebCore/Modules/webxr/WebXRSession.h: Modified to add EnabledFeatures Property
* Source/WebCore/Modules/webxr/WebXRSession.idl: Modified to add EnabledFeatures Property
* LayoutTests/http/wpt/webxr/xrSession_enabledFeatures.html: Added, Imported from WPT. Modified to not test for anchor support which is unsupported in WebKit.
* LayoutTests/http/wpt/webxr/xrSession_enabledFeatures-expected.txt: Added, Imported from WPT.
* LayoutTests/http/wpt/webxr/resources/webxr_test_constants_single_view.js: Modified, Updated by importing from WPT.

Canonical link: <a href="https://commits.webkit.org/281643@main">https://commits.webkit.org/281643@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/31530d3a2ab1852c59c3781d4913cf92ba98e16b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60494 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39850 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13061 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64417 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11031 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62624 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47526 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11262 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48949 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7674 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62525 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37114 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52403 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29781 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33810 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9644 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9946 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55691 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9925 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66148 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4432 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9752 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56316 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4451 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52376 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56494 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/13471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3676 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35657 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36739 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37830 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36484 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebCore/Modules/webxr/WebXRSession.cpp:
```diff
@@ -110,6 +111,19 @@ const WebXRInputSourceArray& WebXRSession::inputSources() const
+    for (const auto& feature : m_requestedFeatures) {
+        String sessionFeature = PlatformXR::sessionFeatureDescriptor(feature);
+        if (sessionFeature != ""_s)
+            enabledFeatureArray.append(sessionFeature);
```
Can WTFMove the string here since it’s not used after the append call

AdaRoseCannon replied:
Changed, thank you!

AdaRoseCannon committed:
Add the `enabledFeatures` property to WebXR Sessions.
https://bugs.webkit.org/show_bug.cgi?id=277155

Reviewed by Mike Wyrzykowski.

Add the `enabledFeatures` property to WebXR Sessions.

* Source/WebCore/Modules/webxr/WebXRSession.cpp: Modified to add EnabledFeatures Property
* Source/WebCore/Modules/webxr/WebXRSession.h: Modified to add EnabledFeatures Property
* Source/WebCore/Modules/webxr/WebXRSession.idl: Modified to add EnabledFeatures Property
* LayoutTests/http/wpt/webxr/xrSession_enabledFeatures.html: Added, Imported from WPT. Modified to not test for anchor support which is unsupported in WebKit.
* LayoutTests/http/wpt/webxr/xrSession_enabledFeatures-expected.txt: Added, Imported from WPT.
* LayoutTests/http/wpt/webxr/resources/webxr_test_constants_single_view.js: Modified, Updated by importing from WPT.

Canonical link: https://commits.webkit.org/281643@main
